website/docs/components/table/partials/code/how-to-use.md

```diff
@@ -545,6 +545,153 @@ In this case the table layout is still set to `auto` (default). If instead you w
+    @totalItems={{this.demoTotalItems}}
+    @currentPage={{this.demoCurrentPage}}
+    @pageSizes={{array 2 4}}
+    @currentPageSize={{this.demoCurrentPageSize}}
+    @onPageChange={{this.demoOnPageChange}}
+    @onPageSizeChange={{this.demoOnPageSizeChange}}
+    @ariaLabel="Pagination for multi-select table"
+  />
+</div>
+```
+
+Depending on the expected behavior, you will need to implement the consumer-side logic that handles the persistence (or not) using the `@onSelectionChange` callback function. For the example above, something like this:
+
+```javascript
+@action
+demoOnSelectionChangeWithPagination({ selectableRowsStates }) {
+  // we loop over all the displayed table rows (a subset of the dataset)
+  selectableRowsStates.forEach((row) => {
+    // we find the record in the dataset corresponding to the current row
+    const recordToUpdate = this.demoSourceData.find(
+      (modelRow) => modelRow.id === row.selectionKey
+    );
+    if (recordToUpdate) {
+      // we update the record `isSelected` state based on the row (checkbox) state
+      recordToUpdate.isSelected = row.isSelected;
+    }
+  });
+}
+
+```
+
+For details about the arguments provided to the `@onSelectionChange` callback function refer to the [Component API](#component-api) section.
+
+
+#### Usability and accessibility considerations
+
+Since the "selected" state of a row is communicated visually via the checkbox selection and for screen-reader users via the `aria-label` applied to the checkbox, there are some important considerations to keep in mind when implementing a multi-select table.
+
+If the selection status of the rows is persisted even when a row is not displayed in the UI, you need to be mindful of what the expectations of the user may be: how do we make them aware that the action that they are going to perform may involve table rows that were previously selected by the user but that are not displayed in that moment?
+
+Even more complex is the case of the "Select all" checkbox. While it's pretty straightforward the expected behaviour when all the table rows are displayed on the table (precisely, "select all the rows in the table"), it may not be obvious what is the expected behaviour when the table rows are paginated of filtered.
+
+If as a user I am looking at a subset of the whole possible rows, and I click "select all", what should I expect to happen? That all (and only) the displayed rows are selected –and this is what happens in the example above– or that all the data/model rows are selected, even if not displayed at the moment? In the first case, the "select all" state changes whenever the rows are changes (eg. the user chooses a different pagination number), and this could be confusing. In the second case, it may not be obvious to the user that _all_ the data records have been selected, and it may unintentionally perform an action on them under the assumption that they are performing it only on the displayed table rows.
```

Comment:
While I know what you're trying to say here, this is a bit mangled. Maybe it would be better to break this down in a step-by-step sequence or something that isn't a long paragraph. Maybe this can be broken up like:

Consider the experience of a user intending to select all or a subset of all possible rows:

If a user interacts with a "select all" function or button, is the expectation that only displayed rows  are selected (what happens in the example above), or that all of the rows in the data set/model are selected, even if not displayed in the current view? 

In the first scenario, the "select all" state changes depending on what rows are in view and can be confusing.

In the second scenario it might not be obvious that all of the rows have been selected and may result in the user unintentionally performing a destructive action under the assumption that they have only selected the rows in the current view.

---

That isn't really perfect either, but it think it could definitely help to break it up a bit. Though, a case could be made to remove this until we have a better idea of how to talk about it. It's super tricky.